app/src/main/java/com/github/sdpteam15/polyevents/model/User.kt

```diff
@@ -0,0 +1,40 @@
+package com.github.sdpteam15.polyevents.model
+
+import com.google.firebase.firestore.IgnoreExtraProperties
+
+// TODO: builder pattern maybe?
+// TODO: Make username unique, and use googleId to avoid duplicate keys
+
+/**
+ * Entity model for application user
+ *
+ * @property googleId the id of a user when he gets authenticated to google services
+ * (this is unique to each user)
+ * @property userType the type of a user (e.g. admin, staff..). Predefined types are
+ * defined in the companion object
+ * @property username username of a user to use and identify him in the application
+ * @property name fullname of a user
+ * @property age the user's age
+ * @property displayName the name that will be displayed on a user's profile and other activites
+ * @property email the email of the user (by default google's)
+ */
+@IgnoreExtraProperties
+data class UserEntity (
+    val googleId: String? = null,
+    val userType: String? = null,
+    val username: String? = null,
+    val name: String? = null,
+    val age: Int? = null,

```

Comment:
Shouldn't it be a birth date instead?

Reply:
Yes will add a TODO for this. LocalDateTime makes age calculation based on birthdate easier. But will have to up the minimum SDK version for this. We'll discuss this with the team.